pkg/ping/cluster.go

```diff
@@ -0,0 +1,122 @@
+package ping
+
+import (
+	"context"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/pluralsh/deployment-operator/internal/helpers"
+	"github.com/pluralsh/deployment-operator/pkg/common"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/klog/v2"
+
+	"github.com/Masterminds/semver/v3"
+	"github.com/pluralsh/polly/containers"
+	"github.com/samber/lo"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const clusterPingerName = "cluster pinger"
+
+func RunClusterPingerInBackgroundOrDie(ctx context.Context, pinger *Pinger, duration time.Duration) {
+	klog.Info("starting ", clusterPingerName)
+
+	err := helpers.BackgroundPollUntilContextCancel(ctx, duration, true, true, func(_ context.Context) (done bool, err error) {
```

Comment:
does this automatically jitter?  We definitely want to make sure that's there

Reply:
I have to add the jitter

Reply:
done